README.adoc

```diff
@@ -148,83 +155,84 @@ Write a simple `Consumer` class that consumes the artist JSON using both JSON-P
-
-`start/src/main/java/io/openliberty/guides/consumingrest/service/ArtistResource.java`:
+Write the method for consuming the REST service using JSON-B in the `Consumer` class:
 
 [source, java, indent=0]
 ----
-include::finish/src/main/java/io/openliberty/guides/consumingrest/service/ArtistResource.java[tags=getTotalArtists]
+include::finish/src/main/java/io/openliberty/guides/consumingrest/Consumer.java[tags=consumeWithJsonb]
 ----
 
-=== Using a JSON binding library
-
-Jackson is a JSON library for Java that is used to serialize Java objects to JSON messages and vice versa.
+This method makes a `GET` request to the running artist service and retrieves the JSON.
 
-To include the Jackson JAX-RS JSON provider in your project, add the following dependency to your `pom.xml`,
-which has been done for you:
+To bind the JSON into an `Artist` array, use the `Artist[]` entity type in the `readEntity` call:
 
-[source, xml]
+[source, java, indent = 0, role="no_copy"]
 ----
-<dependency>
-  <groupId>com.fasterxml.jackson.jaxrs</groupId>
-  <artifactId>jackson-jaxrs-json-provider</artifactId>
-  <version>2.9.0</version>
-</dependency>
+Artist[] artists = response.readEntity(Artist[].class);
 ----
 
-Write the method for consuming the REST service using Jackson's JSON binding in the `Consumer` class:
+Write the `getJsonString` and `getTotalAlbums` methods in the `ArtistResource` class. The `getTotalAlbums`
+method will use JSON-B to return the total number of albums present in the JSON for a particular artist
+and -1 if this artist does not exist using JSON-B.  The `getJsonString` method will return the JSON as
+a string that we will use later for testing.
+
+`start/src/main/java/io/openliberty/guides/consumingrest/service/ArtistResource.java`:
 
 [source, java, indent=0]
 ----
-include::finish/src/main/java/io/openliberty/guides/consumingrest/Consumer.java[tags=consumeWithJackson]
+include::finish/src/main/java/io/openliberty/guides/consumingrest/service/ArtistResource.java[tags=getJsonString]
+include::finish/src/main/java/io/openliberty/guides/consumingrest/service/ArtistResource.java[tags=getTotalAlbums]
 ----
 
-To use Jackson's JAX-RS JSON provider, you must explicitly register it like the following line since
-automatic discovery and registration are not available in the JAX-RS client API:
+=== Processing JSON using JSON-P
 
-[source, java]
+Write the method for consuming the REST service using JSON-P in the `Consumer` class:
+
+[source, java, indent=0]
 ----
-client.register(JacksonJsonProvider.class);
+include::finish/src/main/java/io/openliberty/guides/consumingrest/Consumer.java[tags=consumeWithJsonp]
 ----
 
-To bind the JSON into an `Artist` array, use the `Artist[]` entity type in the `readEntity` call:
+Write the `collectArtists` and `collectAlbums` helpers. These methods will parse that JSON and
+collect its objects into individual POJOs.  Notice that we are able to use our custom constructors
```

Comment:
- Remove extra space between sentences on line 213.
- Remove first person pronoun. I'd rewrite as, "You can use the custom constructors to..."